### PR TITLE
feat(Dropdown): make deburr opt-in, and deburr input

### LINKED ISSUE #2163: Dropdown: remove diacritics on filter also converts national letters
✖ CSS ISSUES → Post on https://github.com/Semantic-Org/Semantic-UI

✖ USAGE QUESTIONS → Use these dedicated resources:
      Docs - http://react.semantic-ui.com
      Chat - https://gitter.im/Semantic-Org/Semantic-UI-React
      SO - http://stackoverflow.com/search?q=semantic-ui-react

✔ BUGS → This form is required:
https://github.com/Semantic-Org/Semantic-UI-React/pull/2021
introduced a filter that removes diacritics on the options in dropdowns, this has an unfortunate side effect as it also converts national characters (ex. ø = o). Scandianvian users would never expect to try searching using o instead of ø - so it breaks the feature.

I suggest that deburr is made optional - and is false by default.

### Steps
Create a dropdown that contains the letter ø in one of it's options
Search for 'ø'
(same goes for æåäö etc)

### Expected Result
List the option that contains ø

### Actual Result
Nothing is listed

### Version
0.73.1

### Testcase
[Fork, update, and replace this pen to show the bug]:
http://codepen.io/levithomason/pen/ZpBaJX

### PR DESCRIPTION
**Pull request by patrikmolsson**
WIP pull request. Would gladly appreciate feedback on both tests and source code!

Fixes #2163.

- [x] Make deburr opt-in
- [x] When having deburr, the input string should also be deburred.
- [x] Add proper documentation
- [x] Add example of using deburr
- [x] Add example of custom search function
- [x] Add proper "should not filter diacritics when deburr is not set" test

### CODE REVIEW

**Comment by patrikmolsson** [comment]
@levithomason I think I'm done now, but I would love some input on the prop naming before we merge!

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js
Comment (layershifter) [inline docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Let's sort imports in alphabetical order

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js
Comment (layershifter) [inline docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: And props in alphatical order, too :smile:

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleDeburrSearch.js
Comment (layershifter) [inline docs/app/Examples/modules/Dropdown/Usage/DropdownExampleDeburrSearch.js]: ^

File: src/modules/Dropdown/Dropdown.js
Comment (layershifter) [inline src/modules/Dropdown/Dropdown.js]: Let's sort props there too

**Review by layershifter — CHANGES_REQUESTED** [review]
@patrikmolsson Thanks, awesome work :+1: I've left styling comments, let's fix them

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js
Comment (patrikmolsson) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Of course! :smile: On it!

Is there a linting rule for this that I'm missing?

Comment (layershifter) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Nope, we don't have or don't use it :thinking:

Comment (patrikmolsson) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Alright. Btw, how do you sort it?  Alphabetically by module name, i.e. ```react```, ```lodash```, and ```semantic-ui-react```?

Comment (patrikmolsson) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: --> ```lodash```, ```react```, ```semantic-ui-react```

Comment (layershifter) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Yes, by module name.

**Comment by patrikmolsson** [comment]
@layershifter There it is! 

By the way, what are your thoughts on the prop naming? Is it clear what ```deburr``` means?

**Review by layershifter — APPROVED** [review]

**Comment by layershifter** [comment]
@patrickgaskill `fuzzy` looks better to me, however lodash uses `deburr`. It's a complicated question

**Comment by patrikmolsson** [comment]
@layershifter ```fuzzy``` might also imply that searching for ```ac``` will match ```abc```...

**Comment by patrikmolsson** [comment]
I've thought about this, but cannot really find an appropriate third option.

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js
Comment (levithomason) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: Thanks for updating.  FWIW, there is a fixable lint rule for this that we should turn on [`sort-imports`](https://eslint.org/docs/rules/sort-imports) so this is done automatically for people.

**Comment by levithomason** [comment]
Beautiful PR, I can see the care and attention to detail you put in.  ❤️ 

### Prop name

Regarding the prop name, lodash is the most installed NPM package and it calls this process `deburr`ing.  It is likely many people will recognize that term.  However, there is a formal specification and process for this.

A Unicode character with diacritics is called a "[precomposed character](https://en.wikipedia.org/wiki/Precomposed_character)".  Example, `é` is composed of the letter `e` and `´` (acute accent).  When these characters are separated like this they are "decomposed" (see previous link).

In terms of Unicode equivalence, this is called "[normalization](https://en.wikipedia.org/wiki/Unicode_equivalence#Normalization)".  Java has a class called "[Normalizer](https://docs.oracle.com/javase/7/docs/api/java/text/Normalizer.html)" which performs this operation.  The Unicode Standard includes a [Normalization Chart](http://www.unicode.org/charts/normalization/) which maps these characters to one another.  The technical process by which equivalent Unicode values are determined is called [_decomposition_](http://www.unicode.org/reports/tr15/tr15-23.html#Decomposition).

So, I think we could go with `deburr`, `normalize`, or `decompose`.  I have my opinion, but I'll let you two choose.

***

@layershifter this is good to merge once you guys are settled on the most fitting prop name.

**Comment by layershifter** [comment]
I'm think that we can go with `deburr`.

**Comment by patrikmolsson** [comment]
@levithomason Thanks for the input, and for the kind words! As @layershifter says, let's go with deburr :relaxed:

**Comment by patrickgaskill** [comment]
@layershifter You tagged the wrong Patrick ;)

**Comment by layershifter** [comment]
@levithomason Can you merge this? As `codecov` failed, I don't have enough rights :smile:

File: docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js
Comment (patrikmolsson) [reply docs/app/Examples/modules/Dropdown/Usage/DropdownExampleCustomSearchFunction.js]: @levithomason It seems that lint does not sort on the module name, but rather on the imported member, or am I missing something?

**Comment by levithomason** [comment]
Sorry folks, Codecov is having issues finding a base to compare some PRs against and incorrectly failing statuses.

This will go out in the next release, today or tomorrow.

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.76.0`

**Pull request merged**
